test/adapters/optimisticAuctionRebalanceExtension.spec.ts

```diff
@@ -0,0 +1,757 @@
+
+    await setV2Setup.integrationRegistry.addIntegration(
+      setV2Setup.auctionModule.address,
+      "ConstantPriceAdapter",
+      priceAdapter.address,
+    );
+
+    useAssetAllowlist = false;
+    allowedAssets = [];
+
+    setToken = await setV2Setup.createSetToken(
+      [setV2Setup.dai.address, setV2Setup.wbtc.address, setV2Setup.weth.address],
+      [ether(100), bitcoin(0.01), ether(0.1)],
+      [setV2Setup.auctionModule.address, setV2Setup.issuanceModule.address],
+    );
+
+    await setV2Setup.issuanceModule.initialize(setToken.address, ADDRESS_ZERO);
+
+    baseManager = await deployer.manager.deployBaseManager(
+      setToken.address,
+      operator.address,
+      methodologist.address,
+    );
+    baseManager = baseManager.connect(operator.wallet);
+    await setToken.setManager(baseManager.address);
+
+    auctionRebalanceExtension = await deployer.extensions.deployOptimisticAuctionRebalanceExtension(
+      baseManager.address,
+      setV2Setup.auctionModule.address,
+      useAssetAllowlist,
+      allowedAssets,
+    );
+    auctionRebalanceExtension = auctionRebalanceExtension.connect(operator.wallet);
+  });
+
+  addSnapshotBeforeRestoreAfterEach();
+
+  describe("#constructor", () => {
+    let subjectBaseManager: Address;
+    let subjectAuctionRebalanceModule: Address;
+      let subjectUseAssetAllowlist: boolean;
+      let subjectAllowedAssets: Address[];
+
+    beforeEach(async () => {
+      subjectBaseManager = baseManager.address;
+      subjectAuctionRebalanceModule = setV2Setup.auctionModule.address;
+        subjectUseAssetAllowlist = false;
+        subjectAllowedAssets = [];
+    });
+
+    async function subject(): Promise<OptimisticAuctionRebalanceExtension> {
+      let extension = await deployer.extensions.deployOptimisticAuctionRebalanceExtension(
+        subjectBaseManager,
+        subjectAuctionRebalanceModule,
+          subjectUseAssetAllowlist,
+          allowedAssets,
+      );
+      extension = extension.connect(operator.wallet);
+      await baseManager.addAdapter(extension.address);
+      return extension;
+    }
+
+    it("should set the correct manager core", async () => {
+      const auctionExtension = await subject();
+
+      const actualBaseManager = await auctionExtension.manager();
+      expect(actualBaseManager).to.eq(subjectBaseManager);
+    });
+
+    it("should set the correct auction rebalance module address", async () => {
+      const auctionExtension = await subject();
+
+      const actualAuctionRebalanceModule = await auctionExtension.auctionModule();
+      expect(actualAuctionRebalanceModule).to.eq(subjectAuctionRebalanceModule);
+    });
+    it("should be able to initialize extension and module at the same time", async () => {
+      const auctionExtension = await subject();
+      await expect(
+        auctionExtension
+          .initialize(),
+      ).to.not.be.reverted;
+    });
```

Comment:
what is this test? https://github.com/IndexCoop/index-coop-smart-contracts/blob/4aa9028018a2ac533b60b55bf01c03185285616b/contracts/adapters/AuctionRebalanceExtension.sol#L187